### ISSUE TITLE
Evaluation does not produce output

### ISSUE THREAD

**Issue by junyang-cynopsis** [nlpyang/BertSum#7]
![Screenshot (2)](https://user-images.githubusercontent.com/45582602/55682532-7b7ffe00-5967-11e9-8a83-812160b8ae82.png)

Hi, I am facing some troubles evaluating. Would you mind telling me what's wrong with my evaluation parameters?

**Comment by nlpyang** [comment]
Hi,

Please make sure you have saved checkpoints in model_path (filenames like model_step_1000.pt).

**Comment by junyang-cynopsis** [comment]
Hi thanks for the pointer, i managed to get the validation running, however it gets stucked at this line, is this the expected outcome?

![image](https://user-images.githubusercontent.com/45582602/55728890-a2156600-5a47-11e9-89d8-2e44ed685f44.png)

**Comment by junyang-cynopsis** [comment]
![Untitled](https://user-images.githubusercontent.com/45582602/55728936-bc4f4400-5a47-11e9-9b7d-e0e45103d801.png)
Hi thanks for the pointer, i managed to get the validation running. However, I'm now stucked at this line, is the expected?

**Comment by nlpyang** [comment]
Hi,
It is not stuck. If you check cpu usage, you can see it running. Although, with cpu, it will be very slow.

**Comment by zsun1029** [comment]
Hi, I have encountered a similar problem, in training step, with my own dataset, with gpu. 

my command is ：
python train.py -mode train -encoder transformer -dropout 0.1 -lr 2e-3 -visible_gpus 0,1 -gpu_ranks 0,1 -bert_data_path ../bert_data -model_path ../models/bert_transformer -world_size 3 -report_every 50 -save_checkpoint_steps 1000 -batch_size 3000 -decay_method noam -train_steps 50000 -accum_count 2 -log_file ../logs/bert_transformer.log -use_interval true -warmup_steps 10000

and I'm stucked at this line：
![image](https://user-images.githubusercontent.com/46882246/55771596-44226600-5abb-11e9-8035-69c5c0c31e73.png)
and another program with the same command stuck at this line:
![image](https://user-images.githubusercontent.com/46882246/55771630-6b793300-5abb-11e9-9bbb-c2f6f479634d.png)

and I check the Program status: 
![image](https://user-images.githubusercontent.com/46882246/55771772-ffe39580-5abb-11e9-9e4b-813db0ea0322.png)
another program status:
![image](https://user-images.githubusercontent.com/46882246/55771833-2bff1680-5abc-11e9-87a3-841337515ccf.png)

they're all sleeping, T-T, Do you know how to solve it?
Thanks in advance T-T

**Comment by nlpyang** [comment]
When you are using two gpus, -world_size should be 2

**Comment by zsun1029** [comment]
Oh yes Yesterday I found my mistake in the first one, but I want to see if the program will report an error so I did not kill it.

but the second command is：
python train.py -mode train -encoder transformer -dropout 0.1 -lr 2e-3 -visible_gpus 1,2,3  -gpu_ranks 2,1,3 -bert_data_path ../bert_data -model_path ../models/bert_transformer2 -world_size 3 -report_every 50 -save_checkpoint_steps 1000 -batch_size 3000 -decay_method noam -train_steps 50000 -accum_count 2 -log_file ../logs/2.log -use_interval true -warmup_steps 10000

it has the same status too, also sleep 
![image](https://user-images.githubusercontent.com/46882246/55796263-29241600-5afc-11e9-81f0-59db362ca762.png)

and every time I kill it, I find it always stuck in this line:
![image](https://user-images.githubusercontent.com/46882246/55796396-8750f900-5afc-11e9-9646-a59187a78bf6.png)

File "train.py", line 65, in multi_main
    p.join()

**Comment by nlpyang** [comment]
gpu_ranks should be 0,1,2

**Comment by nlpyang** [comment]
I did encounter this problem with earlier version pytorch. You can try different pytorch versions to see if the problem is solved.

**Comment by Shashi456** [comment]
@nlpyang  I have a similar problem, the evaluation doesn't provide any output. 

![image](https://user-images.githubusercontent.com/18056781/56482007-77172180-64df-11e9-9110-b2159147cb8e.png)
